src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,2968 @@
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/utility.h>
+
+#include <xrpld/app/tx/apply.h>
+
+#include <xrpl/protocol/Batch.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/STParsedJSON.h>
+#include <xrpl/protocol/Sign.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/jss.h>
+
+namespace ripple {
+namespace test {
+
+class Batch_test : public beast::unit_test::suite
+{
+    struct TestLedgerData
+    {
+        int index;
+        std::string txType;
+        std::string result;
+        std::string txHash;
+        bool isInner;
+    };
+
+    struct TestBatchData
+    {
+        std::string result;
+        std::string txHash;
+    };
+
+    struct TestSignData
+    {
+        int index;
+        jtx::Account account;
+    };
+
+    Json::Value
+    getTxByIndex(Json::Value const& jrr, int const index)
+    {
+        for (auto const& txn : jrr[jss::result][jss::ledger][jss::transactions])
+        {
+            if (txn[jss::metaData][sfTransactionIndex.jsonName] == index)
+                return txn;
+        }
+        return {};
+    }
+
+    void
+    validateBatch(
+        jtx::Env& env,
+        TxID const& parentBatchId,
+        std::vector<TestBatchData> const& batchResults,
+        int const& pbIndex = 0)
+    {
+        Json::Value params;
+        params[jss::ledger_index] = env.current()->seq() - 1;
+        params[jss::transactions] = true;
+        params[jss::expand] = true;
+        auto const jrr = env.rpc("json", "ledger", to_string(params));
+
+        // Validate the number of transactions in the ledger
+        auto const transactions =
+            jrr[jss::result][jss::ledger][jss::transactions];
+        BEAST_EXPECT(transactions.size() == batchResults.size() + 1 + pbIndex);
+
+        // Validate ttBatch is correct index
+        auto const txn = getTxByIndex(jrr, pbIndex);
+        BEAST_EXPECT(txn.isMember(jss::metaData));
+        Json::Value const meta = txn[jss::metaData];
+        BEAST_EXPECT(txn[sfTransactionType.jsonName] == "Batch");
+        BEAST_EXPECT(meta[sfTransactionResult.jsonName] == "tesSUCCESS");
+
+        // Validate the inner transactions
+        for (TestBatchData const& batchResult : batchResults)
+        {
+            Json::Value jsonTx;
+            jsonTx[jss::binary] = false;
+            jsonTx[jss::transaction] = batchResult.txHash;
+            jsonTx[jss::id] = 1;
```

Comment:
I don't think this line is needed?

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5060/commits/d1bf0f6d83acf054b5ef484788369ffd560603da